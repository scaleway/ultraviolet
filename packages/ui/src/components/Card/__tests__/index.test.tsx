import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Card } from '..'

describe('card', () => {
  test('renders correctly with header', () =>
    shouldMatchEmotionSnapshot(<Card header="Title">Hello</Card>))

  test('renders correctly with advanced header', () =>
    shouldMatchEmotionSnapshot(
      <Card header={<h2>Advanced Title</h2>}>Hello</Card>,
    ))
  test('renders correctly with advanced subHeader', () =>
    shouldMatchEmotionSnapshot(
      <Card subHeader={<h2>Advanced subHeader</h2>}>Hello</Card>,
    ))
  test('renders correctly with subHeader', () =>
    shouldMatchEmotionSnapshot(<Card subHeader="Title">Hello</Card>))

  test('renders correctly with header and subHeader', () =>
    shouldMatchEmotionSnapshot(
      <Card header="Main title" subHeader="Title">
        Hello
      </Card>,
    ))

  test('renders correctly with advanced header and subHeader', () =>
    shouldMatchEmotionSnapshot(
      <Card
        header={<h2>Advanced Title</h2>}
        subHeader={<h2>Advanced subHeader</h2>}
      >
        Hello
      </Card>,
    ))

  test('renders correctly without header', () =>
    shouldMatchEmotionSnapshot(<Card>Hello</Card>))

  test('renders correctly with disabled', () =>
    shouldMatchEmotionSnapshot(<Card disabled>Hello</Card>))

  test('renders correctly with disabled and header', () =>
    shouldMatchEmotionSnapshot(
      <Card disabled header="Title">
        Hello
      </Card>,
    ))

  test('renders correctly with active', () =>
    shouldMatchEmotionSnapshot(<Card active>Hello</Card>))

  test('renders correctly with active and disabled', () =>
    shouldMatchEmotionSnapshot(
      <Card active disabled>
        Hello
      </Card>,
    ))

  test('renders correctly with className', () =>
    shouldMatchEmotionSnapshot(<Card className="test">Hello</Card>))

  test('renders correctly with data-testid', () =>
    shouldMatchEmotionSnapshot(<Card data-testid="test">Hello</Card>))
})
