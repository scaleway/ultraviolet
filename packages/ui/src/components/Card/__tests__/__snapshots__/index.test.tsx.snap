// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`card > renders correctly with active 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="true"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with active and disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="true"
      data-is-active="true"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with advanced header 1`] = `
<DocumentFragment>
  .emotion-0[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-2 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-2[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-2[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
      data-disabled="false"
      style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
    >
      <h2>
        Advanced Title
      </h2>
      <div
        class="emotion-2 emotion-3"
        data-disabled="false"
        data-is-active="false"
      >
        Hello
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with advanced header and subHeader 1`] = `
<DocumentFragment>
  .emotion-0[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-2 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-2[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-2[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
      data-disabled="false"
      style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
    >
      <h2>
        Advanced Title
      </h2>
      <div
        class="emotion-2 emotion-3"
        data-disabled="false"
        data-is-active="false"
      >
        <div
          class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_1rem_xxsmall__toi52ud styles_justifyContent_normal_xxsmall__toi52u5d"
          style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
        >
          <h2>
            Advanced subHeader
          </h2>
          Hello
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with advanced subHeader 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="false"
    >
      <div
        class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_1rem_xxsmall__toi52ud styles_justifyContent_normal_xxsmall__toi52u5d"
        style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
      >
        <h2>
          Advanced subHeader
        </h2>
        Hello
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with className 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="test emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="false"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with data-testid 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="false"
      data-testid="test"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="true"
      data-is-active="false"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with disabled and header 1`] = `
<DocumentFragment>
  .emotion-0[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-2 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-2[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-2[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
      data-disabled="true"
      style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
    >
      <h2
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_strong__m4c9owj style_variant_heading__m4c9ow11 style_disabled_true__m4c9ow1a style_undefined_compound_61__m4c9ow31"
      >
        Title
      </h2>
      <div
        class="emotion-2 emotion-3"
        data-disabled="true"
        data-is-active="false"
      >
        Hello
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with header 1`] = `
<DocumentFragment>
  .emotion-0[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-2 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-2[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-2[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
      data-disabled="false"
      style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
    >
      <h2
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_strong__m4c9owj style_variant_heading__m4c9ow11 style_disabled_false__m4c9ow1b style_undefined_compound_25__m4c9ow21 style_undefined_compound_73__m4c9ow3d"
      >
        Title
      </h2>
      <div
        class="emotion-2 emotion-3"
        data-disabled="false"
        data-is-active="false"
      >
        Hello
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with header and subHeader 1`] = `
<DocumentFragment>
  .emotion-0[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-2 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-2[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-2[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
      data-disabled="false"
      style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
    >
      <h2
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_strong__m4c9owj style_variant_heading__m4c9ow11 style_disabled_false__m4c9ow1b style_undefined_compound_25__m4c9ow21 style_undefined_compound_73__m4c9ow3d"
      >
        Main title
      </h2>
      <div
        class="emotion-2 emotion-3"
        data-disabled="false"
        data-is-active="false"
      >
        <div
          class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_1rem_xxsmall__toi52ud styles_justifyContent_normal_xxsmall__toi52u5d"
          style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
        >
          <h3
            class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_headingSmallStrong__m4c9ow16 style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
          >
            Title
          </h3>
          Hello
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly with subHeader 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="false"
    >
      <div
        class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_1rem_xxsmall__toi52ud styles_justifyContent_normal_xxsmall__toi52u5d"
        style="--_4k0ekn0: var(--_4k0ekn0); --_4k0ekn1: var(--_4k0ekn1); --_4k0ekn2: var(--_4k0ekn2); --_4k0ekn3: var(--_4k0ekn3);"
      >
        <h3
          class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_headingSmallStrong__m4c9ow16 style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        >
          Title
        </h3>
        Hello
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`card > renders correctly without header 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 1.5rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-0[data-is-active='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      data-disabled="false"
      data-is-active="false"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;
